### PR TITLE
Handle re-authing when you lose session, and/or change password

### PR DESCRIPTION
**Pull request by orta**
Because I changed my password, I couldn't do things like sync or change availability. Now I find out why. Mostly just a port of the code I wrote for Eigen.

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Needs testing on a Phone if change is non-trivial</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by alloy** [comment]
LGTM 👍, just one question about possible alert copy improvement.

**Comment by orta** [comment]
Cool, you didn't publish the review probably ^

File: Classes/Util/App/ARAuthValidator.m [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    }];
+}
+
++ (void)logout
+{
+    UIAlertController *alert = [UIAlertController alertControllerWithTitle:@"Session expired" message:@"Please log in to continue" preferredStyle:UIAlertControllerStyleAlert];
+    UIAlertAction *defaultAction = [UIAlertAction actionWithTitle:@"Continue" style:UIAlertActionStyleDestructive handler:^(UIAlertAction *action) {
+
+        ARAppDelegate *appDelegate = (id)[UIApplication sharedApplication].delegate;
+        [appDelegate startLogout];
```
[the remark this answers is not in the document]
Comment (alloy) [reply Classes/Util/App/ARAuthValidator.m]: Bump

[the remark this answers is not in the document]
Comment (orta) [reply Classes/Util/App/ARAuthValidator.m]: It does that in that function - yep 

Screen Shot 2019-03-23 at 7 15 42 AM

**Pull request merged**